### ISSUE TITLE
Use the chapter-label and part-label attributes, if set

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#2738]
If the chapter-label and part-label attributes are set, and the corresponding numbering attributes are set (sectnums and partnums), prepend the value of these attributes to the chapter and part title, respectively, in the HTML output. This change should be applied in both the main content area and the table of contents.

Example:

```
= Book Title
:doctype: book
:sectnums:
:partnums:
:chapter-label: Chapter
:part-label: Part

= First Part

== First Chapter
```

Should produce:

```
Part I. First Part

Chapter 1. First Chapter
```

**Comment by andrewcarver** [comment]
Will there be some (new?) attribute or other means of control, over whether chapter-numbers restart in each new Part? (See, e.g., http://docbook.sourceforge.net/release/xsl/1.75.2/doc/fo/label.from.part.html.) If that can be added in a clean and reasonably doable way, I think that would be much preferable than arbitrarily forcing one choice or other on the user. (Out there in the real world, there are plenty of books that do it the one way, and plenty that do it the other.)

  If we don't already have such a facility of control in this or previous milestones, and can't add it right away, seems like there ought to be some discussion about which assumption to force (hopefully only temporarily) on the user... though personally, I don't immediately see what the argument would be for EITHER of the assumptions...

**Comment by andrewcarver** [comment]
I notice that in DocBook XSL, the default is that chapter-numbers do NOT restart in each part. Maybe that would be a very slight argument in favor of implementing that assumption (if there's an immediate choice).

**Comment by andrewcarver** [comment]
One other parameter DocBook XSL provides is also pertinent to Parts-numbering:
"Do component labels include the part label?"
http://docbook.sourceforge.net/release/xsl/1.75.2/doc/fo/component.label.includes.part.label.html

This parallels, of course, the question "Do section labels include the component label?"--concerning which Asciidoctor seems currently to answer "Yes".

Shouldn't the user be given control over these, too--especially the first of these? "This feature is most useful when the label.from.part parameter is turned on. In that case, there would be more than one chapter “1”, and the extra part label prefix will identify each chapter unambiguously."

**Comment by mojavelinux** [comment]
> Will there be some (new?) attribute or other means of control, over whether chapter-numbers restart in each new Part?

That's #979. Let's continue that discussion there.

**Comment by mojavelinux** [comment]
Although I suggested part-label and chapter-label, I recalled we had made the decision when designing the xrefstyle feature to name these part-sig and chapter-sig (to be parallel with part-refsig and chapter-refsig). sig is short for signifier. (We could also write it out as part-signifier and chapter-signifier). The reason for this change is that "label" is easily confused with an "id" in latex. We say signifier because it provides a hint of what the type of section is, but is otherwise just embellishment.
